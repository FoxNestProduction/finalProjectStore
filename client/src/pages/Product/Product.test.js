import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import ProductPage from './Product';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: jest.fn(),
  // useParams: () => ({
  //   hash: '123',
  // }),
}));
jest.mock('@mui/material', () => ({
  ...jest.requireActual('@mui/material'),
  useMediaQuery: jest.fn(),
}));

describe('ProductPage Component', () => {
  const dish = {
    // name: 'Product',
    // description: 'About Product',
    // itemNo: '123',
    // currentPrice: 10.99,
    // isTrending: true,
    // rating: 4,
    // imageUrl: 'image.jpg',
    // isSupreme: false,
    // isHealthy: true,
    _id: '123',
  };

  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce([]);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    jest.spyOn(require('react-router'), 'useParams').mockReturnValueOnce({ itemNo: '123' });
  });

  test('should renders ProductPage component', () => {
    // const { asFragment } = render(
    //   <MemoryRouter initialEntries={['/products/:itemNo']}>
    //     <Routes>
    //       <Route path="/products/:itemNo" element={<ProductPage dish={dish} />} />
    //     </Routes>
    //   </MemoryRouter>
    // );

    // expect(asFragment()).toMatchSnapshot();
  });
});
