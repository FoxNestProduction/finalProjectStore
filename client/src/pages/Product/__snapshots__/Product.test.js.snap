// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage Component should renders ProductPage component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-1jhiief-MuiContainer-root"
    >
      <div
        class="container"
      >
        <div
          class="oneProductPageCardSkeleton"
        >
          <div
            class="oneProductPageImageSkeleton loading"
          />
          <div
            class="oneProductPageDescriptionWrapper"
          >
            <div
              class="oneProductPageRaitingSkeleton loading"
            />
            <div
              class="oneProductPageRaitingSkeleton loading"
            />
            <div
              class="oneProductPageDescription loading"
            />
            <div
              class="oneProductPagePriceSkeleton loading"
            />
          </div>
        </div>
      </div>
    </section>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-sjcq8o-MuiContainer-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-ys7d9m-MuiTypography-root"
      >
        Popular
      </h2>
      <div
        class="MuiGrid-root MuiGrid-container css-rs1phg-MuiGrid-root"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-7uovs1-MuiDivider-root"
      />
    </div>
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-83rf2v-MuiContainer-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h5 css-2mz86v-MuiTypography-root"
      >
        Frequently Asked
        <br />
        Questions
      </h3>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How long does delivery take?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How Does It Work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How does your food delivery service work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  What payment options do you accept?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  Do you offer discounts or promotions?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;

exports[`ProductPage Component should renders ProductPage component isLgTablet 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-1jhiief-MuiContainer-root"
    >
      <div
        class="container"
      >
        <div
          class="oneProductPageCardSkeleton"
        >
          <div
            class="oneProductPageImageSkeleton loading"
          />
          <div
            class="oneProductPageDescriptionWrapper"
          >
            <div
              class="oneProductPageRaitingSkeleton loading"
            />
            <div
              class="oneProductPageRaitingSkeleton loading"
            />
            <div
              class="oneProductPageDescription loading"
            />
            <div
              class="oneProductPagePriceSkeleton loading"
            />
          </div>
        </div>
      </div>
    </section>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-sjcq8o-MuiContainer-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-ys7d9m-MuiTypography-root"
      >
        Popular
      </h2>
      <div
        class="MuiGrid-root MuiGrid-container css-rs1phg-MuiGrid-root"
      />
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-7uovs1-MuiDivider-root"
      />
    </div>
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-83rf2v-MuiContainer-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h5 css-2mz86v-MuiTypography-root"
      >
        Frequently Asked
        <br />
        Questions
      </h3>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How long does delivery take?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How Does It Work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How does your food delivery service work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  What payment options do you accept?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  Do you offer discounts or promotions?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;
